improvements to generation parameters

wbruna opened the pull request.

wbruna committed:
tweak max steps value

The 500 value seems to be a leftover from early SDXL
days; adjust it to 150 to match the 3x increase in max
image dimensions.

wbruna committed:
set default step count to 20 to match Koboldcpp.py

wbruna committed:
consistently use the same default parameters on multi generations

LostRuins commented on src/stores/generator.ts:
```diff
@@ -239,17 +239,18 @@ export const useGeneratorStore = defineStore("generator", () => {
+                        let origseed:number = parseInt((params.value.seed).toString());
+                        if (isNaN(origseed) || origseed < 0)
+                        {
+                            origseed = Math.abs((Math.random() * 2 ** 32) | 0);
```
Instead of `Math.abs((Math.random() * 2 ** 32) | 0)`, let's just use a random integer between 1 and 999999. 

The seed will potentially be incremented an unknown number of times, and we don't want to risk overflow or precision errors (in JS, everything is a float).

wbruna replied:
Alright. But we already have that issue, then, because I copied that code from the 'Randomize!' button...

wbruna replied:
I added code to keep it always inside [0, 2**31), and tested extreme values for the increment and wrap-around behavior.

wbruna committed:
refactor seed generation into a function

wbruna committed:
generate random seeds directly on the interface

So the Text2img buttons will reuse the same seed.

wbruna committed:
default to empty string on the seed parameter

Make it display the message "Enter seed here" by default,
instead of -1 (which is still accepted if provided
manually).

wbruna committed:
prevent seed overflow

- the server could consider a large seed as negative, so
force all seeds into the [0, 2**31) range

- avoid a corner case in the random seed function:
`Math.random() * 2 ** 32` could return -2**31, and 2**31
doesn't fit on a 32-bit signed integer

- fixed the metadata display to consider 0 as a valid
seed (easy to happen with very large numerical seeds)

wbruna commented:
From what I could find, a number could go up to 2^53-1 before the float precision became an issue. But the 32 bit integer overflow would happen earlier (at 2^31-1); and the current _server_ can overflow to a negative value, which would prevent a generation from being reproduced (which could happen for manual input). I'll take a look at the server code afterwards.

I've checked the wrap-around behavior with a 0x7ffffffa seed in a 10 image batch: both the increment and wrap-around to 0 seem to work as expected.

I don't think I have any other low-hanging fruits for this PR (the next thing on my radar would be scheduler support, which would need a dedicated PR anyway), so it should be ready to go.
